cmd/kube-apiserver/app/server.go

```diff
@@ -250,15 +253,30 @@ func Run(s *options.APIServer) error {
 		authorizationConfig.RBACClusterRoleBindingRegistry = clusterrolebinding.NewRegistry(clusterrolebindingetcd.NewREST(mustGetRESTOptions("clusterrolebindings")))
 	}
 
-	authorizer, err := authorizer.NewAuthorizerFromAuthorizationConfig(authorizationModeNames, authorizationConfig)
+	apiAuthorizer, err := authorizer.NewAuthorizerFromAuthorizationConfig(authorizationModeNames, authorizationConfig)
 	if err != nil {
 		glog.Fatalf("Invalid Authorization Config: %v", err)
 	}
 
 	admissionControlPluginNames := strings.Split(s.AdmissionControl, ",")
-	client, err := s.NewSelfClient()
+	client, token, err := s.NewSelfClient()
 	if err != nil {
 		glog.Errorf("Failed to create clientset: %v", err)
+	} else if len(token) > 0 {
+		// Set up a loop back authenticator and authorizer
+		var uid = uuid.NewRandom().String()
+		tokenAuthenticator, err := authenticator.NewAuthenticatorFromToken(token, "apiserver", uid, "")
```

Comment:
I personally would probably reverse the flow, that is, make a self-authenticator that returns a token and pass the token into the NewSelfClient method.